clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -1572,10 +1564,10 @@ public void testManualAssignmentChangeWithAutoCommitEnabled(GroupProtocol groupP
 
         ConsumerPartitionAssignor assignor = new RangeAssignor();
 
+        client.prepareResponseFrom(FindCoordinatorResponse.prepareResponse(Errors.NONE, groupId, node), node);
```

Comment:
Why return a non-controller node in `FindCoordinatorResponse`? Is that expected?

Reply:
I revert this change, but this is expected. `CoordinatorRequestManager#onSuccessfulResponse` uses `Integer.MAX_VALUE - coordinator.nodeId()` as coordinator id.

https://github.com/apache/kafka/blob/0ada8fac6869cad8ac33a79032cf5d57bfa2a3ea/clients/src/main/java/org/apache/kafka/clients/consumer/internals/CoordinatorRequestManager.java#L148-L161

Reply:
ok, that was a trick to enable underlay network to handle connection for the the node having different rules